src/hooks/useTimeRemainingInBatch.ts

```diff
@@ -1,15 +1,35 @@
 import useSafeState from './useSafeState'
-import { getSecondsRemainingInBatch } from 'utils'
+import { getTimeRemainingInBatch } from 'utils'
 import { useEffect, useRef } from 'react'
 import { BATCH_TIME } from 'const'
 
 export function useTimeRemainingInBatch(): number {
-  const [timeRemaining, setTimeRemaining] = useSafeState(getSecondsRemainingInBatch())
+  const [timeRemaining, setTimeRemaining] = useSafeState(getTimeRemainingInBatch())
 
   useEffect(() => {
-    const interval = setInterval(() => setTimeRemaining(getSecondsRemainingInBatch()), 1000)
+    let interval: null | NodeJS.Timeout = null
 
-    return (): void => clearInterval(interval)
+    function updateImmediatelyAndStartInterval(): void {
+      // update once
+      setTimeRemaining(getTimeRemainingInBatch())
+      // update every second from now on
+      interval = setInterval(() => setTimeRemaining(getTimeRemainingInBatch()), 1000)
+    }
+
+    // timeout to start the timer exactly at half second
+    const intervalStart = Date.now() % 500
+
+    if (intervalStart === 0) {
+      // to avoid possible scheduling delays, execute right now if exactly at 500ms mark
+      updateImmediatelyAndStartInterval()
+    } else {
+      // otherwise, schedule starting
+      interval = setTimeout(updateImmediatelyAndStartInterval, intervalStart)
```

Comment:
Can you clearInterval using a `timeoutId`??

I mean, if this component is unmounted before the timeout fires, you are calling clearInterval with a timeoutId

Reply:
edit: I learned something today, JS is very weird. Maybe add a comment in line 31 to say, that clears the interval that can contain the intervalId or the timeoutId

![image](https://user-images.githubusercontent.com/2352112/79794590-d4dc8480-8352-11ea-8b80-85b718546024.png)

Reply:
So [I can do it](https://github.com/gnosis/dex-react/blob/d18b43ac14145c1a3ff00a5a1b25f9dfd552f81c/src/hooks/useTimeRemainingInBatch.ts#L90-L92) and none bats an eye, but when @alfetopito does you notice :rofl:

Reply:
> So [I can do it](https://github.com/gnosis/dex-react/blob/d18b43ac14145c1a3ff00a5a1b25f9dfd552f81c/src/hooks/useTimeRemainingInBatch.ts#L90-L92) and none bats an eye, but when @alfetopito does you notice rofl

No one cares about you @Velenir 

OR

No one trusts my code

:thinking:

Reply:
It just depend on the number of cups of coffee I had ☕️☕️☕️☕️☕️☕️☕️☕️